backend/WebApi/Controllers/FoodController.cs

```diff
@@ -27,28 +28,27 @@ public FoodController(IFoodService foodService)
         /// <summary>
         /// Retrieve all food items.
         /// </summary>
-        /// <param name="sortOrder">Optional order by which the food should be sorted</param>
+        /// <param name="foodParameters">Page number and size</param>
         /// <returns></returns>
         [HttpGet]
         [ProducesResponseType((int)HttpStatusCode.OK, Type = typeof(IEnumerable<FoodResponse>))]
         [ProducesResponseType((int)HttpStatusCode.BadRequest)]
-        public IActionResult GetAll(string sortOrder = null)
+        public IActionResult GetAll([FromQuery] FoodParameters foodParameters)
         {
-            if (sortOrder != null)
+            try
             {
-                try
-                {
-                    InputValidator.ValidateFoodSortOrder(sortOrder);
-                }
-                catch (ArgumentException e)
-                {
-                    return BadRequest(e.Message);
-                }
+                InputValidator.ValidateFoodSortOrder(foodParameters.SortOrder);
+            }
+            catch (ArgumentException e)
+            {
+                return BadRequest(e.Message);
             }
 
-            var foodsResp = _foodService.GetAllFood().Select(food => FoodResponse.FromEntity(food));
+            var pagedFoodList = _foodService.GetAllFood(foodParameters);
+
+            this.AddPaginationMetadata(pagedFoodList);
 
-            foodsResp = EntitySorter.SortFoods(foodsResp, sortOrder);
```

Comment:
Where do we sort by sortOrder now?

Reply:
We don't. I couldn't figure out a performant solution using IQueryable. I think we will need to load all data to memory and then sort it. But I didn't implement it yet.